pkg/virt-handler/vm.go

```diff
@@ -291,16 +291,32 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 				domainInterfaceStatusByMac[domainInterfaceStatus.Mac] = domainInterfaceStatus
 			}
 
+			existingInterfacesSpecByName := map[string]v1.Interface{}
+			for _, existingInterfaceSpec := range vmi.Spec.Domain.Devices.Interfaces {
+				if existingInterfaceSpec.Name != "" {
+					existingInterfacesSpecByName[existingInterfaceSpec.Name] = existingInterfaceSpec
+				}
+			}
+
 			// Iterate through all domain.Spec interfaces
 			for _, domainInterface := range domain.Spec.Devices.Interfaces {
 				interfaceMAC := domainInterface.MAC.MAC
 				var newInterface v1.VirtualMachineInstanceNetworkInterface
+				var isMasqueredeInterface bool = false
+
+				if existingInterfacesSpecByName[domainInterface.Alias.Name].Masquerade != nil {
+					isMasqueredeInterface = true
+				}
 
 				if existingInterface, exists := existingInterfaceStatusByName[domainInterface.Alias.Name]; exists {
 					// Reuse previously calculated interface from vmi.Status.Interfaces, updating the MAC from domain.Spec
 					// Only interfaces defined in domain.Spec are handled here
 					newInterface = existingInterface
-					newInterface.MAC = interfaceMAC
```

Comment:
@vatsalparekh Where do you update the MAC address for masquerade-enabled interfaces?

Reply:
@enp0s3 in this PR I'm not updating it yet at any place, just letting it be what it was previously
But I think it should be done at controller level, somewhere in this https://github.com/kubevirt/kubevirt/pull/2963